perun/postprocess/regressogram/methods.py

```diff
@@ -0,0 +1,135 @@
+"""
+Module with regressogram computational method and auxiliary methods at executing of this method.
+"""
+
+import numpy as np
+import numpy.lib.function_base as np_bin_selectors
+from click import BadParameter
+from scipy import stats
+from sklearn import metrics
+
+from perun.postprocess.regression_analysis.tools import validate_dictionary_keys
+
+# required arguments at regressogram post-processor
+_REQUIRED_KEYS = ['bins', 'statistic']
+
+
+def compute(data_gen, configuration):
+    """
+    The regressogram wrapper to execute the analysis on the individual chunks of resources.
+
+    :param iter data_gen: the generator object with collected data (data provider generators)
+    :param dict configuration: the perun and option context
+    :return: list of dict: the computation results
+    """
+    # checking the presence of specific keys in individual methods
+    validate_dictionary_keys(configuration, _REQUIRED_KEYS, [])
+
+    # list of result of the analysis
+    analysis = []
+    for chunk in data_gen:
```

Comment:
Fix: extract chunk[0], chunk[1], chunk[2] into appropriate variable names. If they are tuple you can use "for a, b, c in data_gen" and it should extract correctly.